### PR TITLE
Document where to get filter plugins

### LINKED ISSUE #1606: Document how to use/link to hdf5plugin
[hdf5plugin](https://github.com/silx-kit/hdf5plugin) provides a number of additional compression filters. We should add these filters to those listed at https://h5py.readthedocs.io/en/stable/high/dataset.html#filter-pipeline, and document them appropriately.

This to me seems like the best solution for #611.

### PR DESCRIPTION
**Pull request by takluyver**
Inc. hdf5plugin (the Python package), HDF group's package of plugins, and the index of plugins.

Closes #1606.

### CODE REVIEW

File: docs/high/dataset.rst [hunk 1]
```diff
@@ -261,6 +261,18 @@ dynamically loaded by the underlying HDF5 library. This is done by passing a
 filter number to :meth:`Group.create_dataset` as the ``compression`` parameter.
 The ``compression_opts`` parameter will then be passed to this filter.
 
+.. seealso::
+
+   `hdf5plugin <https://pypi.org/project/hdf5plugin/>`_
+     A Python package with several popular filters, including Blosc and LZ4,
```
Comment (takluyver) [inline docs/high/dataset.rst:267]: Q: which filters to highlight from hdf5plugin? I think Blosc should certainly be mentioned, and I picked LZ4 as the other one I think I'd heard of before learning about HDF5. But maybe it's better to mention ZFP (a lossy filter) to show the breadth. Or maybe it's simpler just to list them all.

cc @kif @vasole @t20100

**Comment by vasole** [comment]
I think ZFP is certainly worth to be mentioned.  It can do wonders with (lossy) compression of floating point data and it is a filter people should consider.

Bitshuffle+LZ4 was the reason we started to provide the module : to support data from some detectors mostly used at synchrotrons but the potential users are mostly already aware of the existence of hdf5plugin :)

If you want to provide some details then I would restrict to Blosc, LZ4 and ZFP as you had envisaged. If you are just going to mention them, then it does not take much more to mention all those already supported.

File: docs/high/dataset.rst (hunk 1 above)
Comment (t20100) [reply docs/high/dataset.rst:267]: I'd mention Blosc as well and ZFP to have a lossy filter.
If you list it all, you probably want to do the same for HDF5 Filter Plugins, and we'll just have to think of updating `h5py` doc when we add a new filter to the zoo.

**Comment by takluyver** [comment]
Thanks both! I'll add ZFP. I'm inclined to give examples rather than a complete list to reduce the risk of it getting out of date, though this isn't a strong preference if you want to argue for a full list.

**Comment by vasole** [comment]
It is fine with me. The reduced list points people to the filters that are really worth to take a closer look at.

**Pull request merged**